appcues/src/main/java/com/appcues/statemachine/StepReference.kt

```diff
@@ -27,6 +33,8 @@ internal sealed class StepReference {
 
     data class StepGroupPageIndex(val index: Int) : StepReference() {
 
+        override val destination = "#$index"
```

Comment:
StepGroupPageIndex is not really being used by the ContinueAction, and if we were to get the index from the getIndex method we would need to have a reference to the experience

Reply:
so this is only ever used by the page swiping gesture right? and thus would never actually show up in a step_interaction analytic for a button?

Reply:
precisely. what should we do here? the only way we would use this is if in the future we change this to add a step_interaction for swipe.

Reply:
Ok, makes sense it _should_ never matter. I think it would be fairly easy to make it work though just to be safe.

* In AppcuesViewModel, where it has UIState.Rendering - add the flatStepIndex in there as another property it tracks.
* in the same file, line 101 where it creates the Rendering state in toRenderingState() - we already have the flatStepIndex, so just pass it though
* line 116 in onPageChanged - we know we're in Rendering state, and we can grab the flatStepIndex and pass it through to StepGroupPageIndex constructor
* now its available as a supplemental property for analytics usage